src/main/java/app/controllers/ProfessorController.java

```diff
@@ -29,4 +45,79 @@ public String professor(Model model)
 
         return "professor";
     }
+
+    @RequestMapping("/new-project")
+    public ModelAndView createNewProject()
+    {
+        return new ModelAndView("new-project","command", new Project());
+    }
+
+
+    // Have to get the professor's ID
+    @RequestMapping(value = "/save",method = RequestMethod.POST)
+    public ModelAndView saveProject(@RequestParam("desc") String description, @RequestParam("rest") String restrictions, @RequestParam("cap") int capacity)
+    {
+        //professor = professorRepository.findById();
+        System.out.println("Description: " + description);
+        System.out.println("Restrictions: " + restrictions);
+        System.out.println("Capacity: " + capacity);
+        List<String> restrictionsList = Arrays.asList(restrictions.split(","));
+
+        professor = professorRepository.findFirstByOrderById();
+        professor.createProject(description, restrictionsList, capacity);
+        professorRepository.save(professor);
+
+        return new ModelAndView("redirect:/professor"/* + professor.getId()*/);
+    }
+
+    @RequestMapping("/new-student/")
+    public ModelAndView addStudentById()
+    {
+        return new ModelAndView("new-student","command", new Student());
+    }
+
+    @RequestMapping(value = "/save-student",method = RequestMethod.POST)
+    public ModelAndView addStudent(@RequestParam("id") int id)
+    {
+        //professor = professorRepository.findById();
+        student = studentRepository.findOne((long) id);
+        professor = professorRepository.findFirstByOrderById();
+
+        professorRepository.save(professor);
+
+        return new ModelAndView("redirect:/professor"/* + professor.getId()*/);
+    }
+
+
+    // Not persisting...have to modify the repository
+    @RequestMapping(value = "professor/archiveProject/{projectID}", method = RequestMethod.GET)
+    public ModelAndView archive(/*@PathVariable(name = "id") String id, */@PathVariable(name = "projectID") String projectID)
```

Comment:
I think `@PathVariable(name = "projectID") String projectID` can be simplified to `@PathVariable("projectID") Long projectID`

Same goes of the rest of the params for the other controller endpoints, you can just specify the type you want.